src/Trip.js

```diff
@@ -1,12 +1,71 @@
 class Trip {
-  constructor(id, userID, destinationID, travelers, date, duration, status, suggestedActivities){
-    this.id = id;
-    this.userID = userID;
-    this.destinationID = destinationID;
-    this.travelers = travelers;
-    this.date = date;
-    this.duration = duration;
-    this.status = status;
-    this.suggestedActivities = suggestedActivities;
+  constructor(tripObj, destinationData){
+    this.id = tripObj.id;
+    this.destinationID = tripObj.destinationID;
+    this.travelers = tripObj.travelers;
+    this.date = tripObj.date;
+    this.duration = tripObj.duration;
+    this.status = tripObj.status;
+    this.destinationObj;
+    this.isCurrent;
+    this.isUpcoming;
+    this.isPast;
+    this.destinationData = destinationData;
+
+    this.checkIfCurrentTrip();
+    this.checkIfUpcomingTrip();
+    this.checkIfPast();
+    this.determineDestination();
```

Comment:
The first 3 lines of these checkif functions look similar- maybe they can be broken into a helper function?